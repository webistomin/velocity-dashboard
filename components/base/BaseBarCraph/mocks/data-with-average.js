export const barGraphDataWithAverage = {
  labels: ['Mon', 'Tue', 'Fri', 'Thu', 'Wed', 'Sat', 'Sun'],
  datasets: [
    {
      label: 'Average',
      borderColor: '#F7C137',
      fill: false,
      lineTension: 0,
      data: [110, 80, 100, 110, 74, 55, 124],
      type: 'line',
    },
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Comfort',
      backgroundColor: '#2E5BFF',
      data: [82, 65, 75, 130, 55, 62, 120],
    },
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Premium',
      backgroundColor: '#8C54FF',
      data: [90, 75, 80, 60, 45, 117],
    },
  ],
};
