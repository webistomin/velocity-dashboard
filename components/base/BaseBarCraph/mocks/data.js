export const barGraphData = {
  labels: ['April', 'May', 'June'],
  datasets: [
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Comfort',
      lineTension: 0.4,
      backgroundColor: '#2E5BFF',
      data: [60, 65, 85],
    },
    {
      barPercentage: 0.5,
      categoryPercentage: 0.3,
      label: 'Premium',
      backgroundColor: '#8C54FF',
      data: [70, 80, 90],
    },
  ],
};
