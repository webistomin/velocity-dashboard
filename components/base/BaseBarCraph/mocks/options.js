export const options = {
  responsive: true,
  maintainAspectRatio: false,
  scales: {
    yAxes: [
      {
        gridLines: {
          color: '#B0BAC9',
          borderDash: [3, 3],
        },
        ticks: {
          suggestedMin: 0,
          suggestedMax: 125,
          fontColor: '#B0BAC9',
        },
      },
    ],
    xAxes: [
      {
        gridLines: {
          color: '#B0BAC9',
          borderDash: [3, 3],
        },
        ticks: {
          fontColor: '#B0BAC9',
        },
      },
    ],
  },
  legend: {
    display: true,
    align: 'end',
    labels: {
      boxWidth: 10,
      usePointStyle: true,
      padding: 20,
      fontColor: '#B0BAC9',
      fontFamily: 'Rubik, system-ui, sans-serif',
    },
  },
  elements: {
    point: {
      radius: 5,
      hoverRadius: 4,
      borderWidth: 3,
      hoverBorderWidth: 3,
      backgroundColor: '#FFF',
    },
  },
};
