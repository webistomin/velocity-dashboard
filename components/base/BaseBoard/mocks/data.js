export const boardData = [
  {
    title: 'Backlog',
    color: 'blue',
    tasks: [
      {
        title: 'Spire1',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire2',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire3',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire4',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'In progress',
    color: 'purple',
    tasks: [
      {
        title: 'Spire5',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire6',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire7',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire8',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'Validate',
    color: 'darkturquoise',
    tasks: [
      {
        title: 'Spire9',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire10',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire11',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire12',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
  {
    title: 'Complete',
    color: 'yellow',
    tasks: [
      {
        title: 'Spire13',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire14',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire15',
        text: 'Tire replacement',
        date: 'June 12',
      },
      {
        title: 'Spire16',
        text: 'Tire replacement',
        date: 'June 12',
      },
    ],
  },
];
