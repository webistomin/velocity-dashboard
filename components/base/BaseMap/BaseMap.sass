+b('base-map')
  &.vue2leaflet-map
    width: calc(100% + 4.8rem)
    position: relative
    left: 50%
    transform: translateX(-50%)
    height: 448px

    .leaflet-marker-icon.marker-cluster
      background-color: var(--color-primary)

      div
        font: var(--text-link)
        color: var(--color-white)
        background-color: inherit

  +e('marker')
    position: relative

    &::before
      content: ''
      width: 8px
      height: 8px
      background-color: red
      +absolute-center()
