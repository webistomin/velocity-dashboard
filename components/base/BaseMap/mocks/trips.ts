export const trips = [
  {
    _id: 'eV8QFGTSp4dywwyCqjsH7',
    __v: 0,
    passengerId: '90xl_i0gwbHaMtUQ4DPBU',
    driverId: 'PA__7xXv46CQWOA1npuYD',
    paymentDetails: {
      type: 'visa',
    },
    trip: {
      type: 'comfort',
      distance: 15,
      approximateTime: 13,
      price: 8,
      startAddress: '290  Eden Drive, Richmond, VA',
      endAddress: '4503  Orchard Street, Richmond, VA',
      startTime: '2020-05-27T14:43:54.403Z',
      endTime: null,
    },
    path: [
      {
        lat: 50.516518,
        lng: 6.993713,
      },
      {
        lat: 50.516566,
        lng: 6.993747,
      },
      {
        lat: 50.516566,
        lng: 6.993747,
      },
      {
        lat: 50.516582,
        lng: 6.993847,
      },
      {
        lat: 50.516574,
        lng: 6.99392,
      },
      {
        lat: 50.516574,
        lng: 6.994007,
      },
      {
        lat: 50.51657,
        lng: 6.994085,
      },
      {
        lat: 50.516551,
        lng: 6.994153,
      },
      {
        lat: 50.516536,
        lng: 6.994224,
      },
      {
        lat: 50.516521,
        lng: 6.994309,
      },
      {
        lat: 50.516507,
        lng: 6.994385,
      },
      {
        lat: 50.516485,
        lng: 6.994459,
      },
      {
        lat: 50.516465,
        lng: 6.994526,
      },
      {
        lat: 50.516444,
        lng: 6.994602,
      },
      {
        lat: 50.516403,
        lng: 6.994652,
      },
      {
        lat: 50.516386,
        lng: 6.994722,
      },
      {
        lat: 50.516333,
        lng: 6.994762,
      },
      {
        lat: 50.516276,
        lng: 6.99481,
      },
      {
        lat: 50.516223,
        lng: 6.994854,
      },
      {
        lat: 50.51617,
        lng: 6.994882,
      },
      {
        lat: 50.516122,
        lng: 6.994915,
      },
      {
        lat: 50.516076,
        lng: 6.994951,
      },
      {
        lat: 50.516028,
        lng: 6.994998,
      },
      {
        lat: 50.515983,
        lng: 6.995041,
      },
      {
        lat: 50.515936,
        lng: 6.99509,
      },
      {
        lat: 50.515889,
        lng: 6.995134,
      },
      {
        lat: 50.515863,
        lng: 6.995193,
      },
      {
        lat: 50.515861,
        lng: 6.995203,
      },
      {
        lat: 50.515902,
        lng: 6.995251,
      },
      {
        lat: 50.515948,
        lng: 6.995238,
      },
      {
        lat: 50.515999,
        lng: 6.995237,
      },
      {
        lat: 50.515962,
        lng: 6.995287,
      },
      {
        lat: 50.515915,
        lng: 6.995296,
      },
      {
        lat: 50.51586,
        lng: 6.995298,
      },
      {
        lat: 50.515815,
        lng: 6.995294,
      },
    ],
    status: 'in-progress',
  },
  {
    _id: 'LSE_bJkAoAsSB0rGP_lfp',
    __v: 1,
    passengerId: '_csR1ydDALnI2crwQ2lTY',
    driverId: 't-wEXo14IEwbseZD4kGxH',
    paymentDetails: {
      type: 'visa',
    },
    trip: {
      type: 'comfort',
      distance: 15,
      approximateTime: 13,
      price: 8,
      endAddress: '290  Eden Drive, Richmond, VA',
      startAddress: '4503  Orchard Street, Richmond, VA',
      startTime: '2020-05-27T14:43:54.403Z',
      endTime: null,
    },
    path: [
      {
        lat: 50.515815,
        lng: 6.995294,
      },
      {
        lat: 50.51586,
        lng: 6.995298,
      },
      {
        lat: 50.515915,
        lng: 6.995296,
      },
      {
        lat: 50.515962,
        lng: 6.995287,
      },
      {
        lat: 50.515999,
        lng: 6.995237,
      },
      {
        lat: 50.515948,
        lng: 6.995238,
      },
      {
        lat: 50.515902,
        lng: 6.995251,
      },
      {
        lat: 50.515861,
        lng: 6.995203,
      },
      {
        lat: 50.515863,
        lng: 6.995193,
      },
      {
        lat: 50.515889,
        lng: 6.995134,
      },
      {
        lat: 50.515936,
        lng: 6.99509,
      },
      {
        lat: 50.515983,
        lng: 6.995041,
      },
      {
        lat: 50.516028,
        lng: 6.994998,
      },
      {
        lat: 50.516076,
        lng: 6.994951,
      },
      {
        lat: 50.516122,
        lng: 6.994915,
      },
      {
        lat: 50.51617,
        lng: 6.994882,
      },
      {
        lat: 50.516223,
        lng: 6.994854,
      },
      {
        lat: 50.516276,
        lng: 6.99481,
      },
      {
        lat: 50.516333,
        lng: 6.994762,
      },
      {
        lat: 50.516386,
        lng: 6.994722,
      },
      {
        lat: 50.516403,
        lng: 6.994652,
      },
      {
        lat: 50.516444,
        lng: 6.994602,
      },
      {
        lat: 50.516465,
        lng: 6.994526,
      },
      {
        lat: 50.516485,
        lng: 6.994459,
      },
      {
        lat: 50.516507,
        lng: 6.994385,
      },
      {
        lat: 50.516521,
        lng: 6.994309,
      },
      {
        lat: 50.516536,
        lng: 6.994224,
      },
      {
        lat: 50.516551,
        lng: 6.994153,
      },
      {
        lat: 50.51657,
        lng: 6.994085,
      },
      {
        lat: 50.516574,
        lng: 6.994007,
      },
      {
        lat: 50.516574,
        lng: 6.99392,
      },
      {
        lat: 50.516582,
        lng: 6.993847,
      },
      {
        lat: 50.516566,
        lng: 6.993747,
      },
      {
        lat: 50.516566,
        lng: 6.993747,
      },
      {
        lat: 50.516518,
        lng: 6.993713,
      },
    ],
    status: 'in-progress',
  },
];
