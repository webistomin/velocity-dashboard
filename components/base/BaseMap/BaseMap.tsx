import { VueComponent } from 'types/vue-components';
import { Component } from 'nuxt-property-decorator';

import './BaseMap.sass';

@Component({
  name: 'BaseMap',
})
export default class BaseMap extends VueComponent {
  url: string = 'https://cartodb-basemaps-{s}.global.ssl.fastly.net/light_all/{z}/{x}/{y}.png';
  travel = [
    [145.0, 175.2],
    [8.3, 218.7],
  ];

  render() {
    return (
      <client-only>
        <L-Map zoom={10} center={[55.751244, 37.618423]} class='base-map'>
          <L-Tile-Layer url={this.url} subdomains='abcd' maxZoom={20} />
          <L-Marker-Cluster
            options={{
              showCoverageOnHover: false,
            }}>
            <L-Marker lat-lng={[55.751244, 37.618423]} />
            <L-Marker lat-lng={[55.851244, 37.618423]} />
            <L-Marker lat-lng={[55.851244, 37.118423]} />
          </L-Marker-Cluster>
          <L-Hotline
            weight={2}
            outline-width={0}
            outline-color={'#2E5BFF'}
            palette={{
              0.0: '#2E5BFF',
              0.5: '#2E5BFF',
              1.0: '#2E5BFF',
            }}
            latLngs={[
              [55.751244, 37.618423, 1],
              [55.851244, 37.618423, 1],
              [55.851244, 37.118423, 1],
            ]}
          />
        </L-Map>
      </client-only>
    );
  }
}
