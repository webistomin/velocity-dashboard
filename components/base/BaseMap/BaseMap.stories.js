import { object } from '@storybook/addon-knobs';
import BaseMap from './BaseMap';
import { trips } from './mocks/trips';

export default {
  title: 'BaseMap',
  parameters: {
    options: {
      showPanel: true,
      selectedPanel: 'storybook/knobs/panel',
    },
  },
};

export const Default = () => ({
  components: { BaseMap },
  props: {
    trips: {
      type: Array,
      default: object('trips', trips),
      required: true,
    },
  },
  template: `<div :style="{height: '100vh'}"><base-map :trips='trips'/></div>`,
});
