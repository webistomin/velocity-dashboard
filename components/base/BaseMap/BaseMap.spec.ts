import { mount } from '@vue/test-utils';
import { SiteThemes } from 'common/types/theme/site-themes';
import BaseMap from './BaseMap';
import { trips } from './mocks/trips';

describe('BaseMap', () => {
  const factory = (propsData = {}, options = {}) => {
    return mount(BaseMap, {
      propsData: {
        trips,
        ...propsData,
      },
      stubs: [
        'nuxt-link',
        'client-only',
        'L-Map',
        'L-Tile-Layer',
        'L-Marker-Cluster',
        'L-Marker',
        'L-Icon',
        'L-Hotline',
        'svg-icon',
      ],
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });

  it('Set max zoom if prop "initialTripId" provided', () => {
    const wrapper = factory({
      initialTripId: 'id',
    });
    const zoom = wrapper.vm.zoomLevel;
    const maxZoom = 18;
    expect(zoom).toEqual(maxZoom);
  });

  it('Emit "markerClick" event on marker click', () => {
    const wrapper = factory();
    const marker = wrapper.find('l-marker-stub');
    marker.trigger('click');
    expect(wrapper.emitted().markerClick);
  });

  it('Call markerClick on marker keydown', () => {
    const wrapper = factory();
    const leafletEvent = {
      originalEvent: {
        key: 'Enter',
      },
    };
    // @ts-ignore
    wrapper.vm.onMarkerKeyDown(leafletEvent, 1);
    expect(wrapper.emitted().markerClick);
  });

  it('Doesnt not call markerClick on marker keydown if key is not enter', () => {
    const wrapper = factory();
    const leafletEvent = {
      originalEvent: {
        key: 'Esc',
      },
    };
    // @ts-ignore
    wrapper.vm.onMarkerKeyDown(leafletEvent, 1);
    expect(wrapper.emitted().markerClick).toEqual(undefined);
  });

  it('Close sidebar on "Esc"', () => {
    const mockedCloseSidebar = jest.fn();
    const wrapper = factory(
      {},
      {
        attachToDocument: true,
        methods: {
          closeSidebar: mockedCloseSidebar,
        },
      }
    );
    wrapper.setData({
      isSidebarVisible: true,
    });
    wrapper.trigger('keydown.esc', { key: 'Escape' });
    expect(mockedCloseSidebar).toHaveBeenCalled();
  });

  it('Delete global event listeners on destroy', () => {
    const mockedCloseSidebar = jest.fn();
    const wrapper = factory(
      {},
      {
        attachToDocument: true,
        methods: {
          closeSidebar: mockedCloseSidebar,
        },
      }
    );
    wrapper.setData({
      isSidebarVisible: true,
    });
    wrapper.destroy();
    wrapper.trigger('keydown.esc', { key: 'Escape' });
    expect(mockedCloseSidebar).not.toHaveBeenCalled();
  });

  it('Set sidebar visibility to false and current id to null on sidebar close', () => {
    const wrapper = factory();
    wrapper.vm.closeSidebar();

    const currentId = wrapper.vm.currentTripId;
    const isSidebarVisible = wrapper.vm.isSidebarVisible;

    expect(currentId).toEqual(null);
    expect(isSidebarVisible).toEqual(false);
  });

  it('Return default icon if no theme provided', () => {
    const wrapper = factory();
    const icon = wrapper.vm.getIcon;
    const result = {
      iconUrl: `/img/map-marker-shelob.svg`,
      iconRetinaUrl: `/img/map-marker-shelob.svg`,
      shadowUrl: '',
      iconSize: [24, 24],
      shadowSize: [0, 0],
      iconAnchor: [12, 12],
    };

    expect(icon).toEqual(result);
  });

  it('Return default primary color if process is not client', () => {
    const wrapper = factory();
    const primaryColor = wrapper.vm.getPrimaryColor;
    const result = 'rgb(46, 91, 255)';

    expect(primaryColor).toEqual(result);
  });

  it('Return color from document if process client', () => {
    (process as any).client = true;
    const wrapper = factory();
    const primaryColor = wrapper.vm.getPrimaryColor;
    const result = 'rgb(46, 91, 255)';
    (process as any).client = false;

    expect(primaryColor).toEqual(result);
  });

  it('Return default theme if process is not client', () => {
    (process as any).client = false;
    const wrapper = factory();
    const currentTheme = wrapper.vm.getCurrentTheme;
    const result = SiteThemes.SHELOB;

    expect(currentTheme).toEqual(result);
  });

  it('Return theme from local storage if process is client', () => {
    (process as any).client = true;
    const wrapper = factory();
    const currentTheme = wrapper.vm.getCurrentTheme;
    const result = null;
    (process as any).client = false;

    expect(currentTheme).toEqual(result);
  });

  it('Return trip info if trip exists', () => {
    const wrapper = factory({
      initialTripId: 'eV8QFGTSp4dywwyCqjsH7',
    });
    const currentTheme = wrapper.vm.getCurrentTripInfo;
    const result = {
      id: 'eV8QFGTSp4dywwyCqjsH7',
      type: 'comfort',
      distance: 15,
      approximateTime: 13,
      price: 8,
      startAddress: '290  Eden Drive, Richmond, VA',
      endAddress: '4503  Orchard Street, Richmond, VA',
    };

    expect(currentTheme).toEqual(result);
  });

  it('Return undefined info if trip does not exists', () => {
    const wrapper = factory({
      initialTripId: '312312312312312312',
    });
    const currentTheme = wrapper.vm.getCurrentTripInfo;
    const result = undefined;

    expect(currentTheme).toEqual(result);
  });

  it('Set null to current id if clicked on same marker', () => {
    const wrapper = factory({
      initialTripId: '1',
    });
    wrapper.vm.onMarkerClick('1');
    expect(wrapper.vm.currentTripId).toEqual(null);
  });
});
