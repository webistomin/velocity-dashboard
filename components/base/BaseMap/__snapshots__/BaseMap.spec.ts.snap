// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseMap Matches snapshot 1`] = `
"<div class=\\"base-map\\">
  <div class=\\"base-map__sidebar \\"><button type=\\"button\\" aria-label=\\"Close\\" class=\\"base-close btn base-map__close\\"></button></div>
  <client-only-stub>
    <l-map-stub zoom=\\"10\\" center=\\"50.516518,6.993713\\" class=\\"base-map__map\\">
      <l-tile-layer-stub url=\\"https://cartodb-basemaps-{s}.global.ssl.fastly.net/light_all/{z}/{x}/{y}.png\\" subdomains=\\"abcd\\" maxzoom=\\"20\\"></l-tile-layer-stub>
      <l-marker-cluster-stub options=\\"[object Object]\\">
        <l-marker-stub lat-lng=\\"[object Object]\\">
          <l-icon-stub iconurl=\\"/img/map-marker-shelob.svg\\" iconretinaurl=\\"/img/map-marker-shelob.svg\\" shadowurl=\\"\\" iconsize=\\"24,24\\" shadowsize=\\"0,0\\" iconanchor=\\"12,12\\"></l-icon-stub>
        </l-marker-stub>
        <l-marker-stub lat-lng=\\"[object Object]\\">
          <l-icon-stub iconurl=\\"/img/map-marker-shelob.svg\\" iconretinaurl=\\"/img/map-marker-shelob.svg\\" shadowurl=\\"\\" iconsize=\\"24,24\\" shadowsize=\\"0,0\\" iconanchor=\\"12,12\\"></l-icon-stub>
        </l-marker-stub>
      </l-marker-cluster-stub>
    </l-map-stub>
  </client-only-stub>
</div>"
`;
